### ISSUE TITLE
Re-enable LLVM and debug assertions for slow builders

### ISSUE THREAD

**Issue by alexcrichton** [rust-lang/rust#59637]
Assertions are currently disabled on a number of builders:

 * test-various - debug asserts, intentional - needed to properly test code size of some tests
 * dist-x86_64-apple - llvm, debug
 * dist-x86_64-apple-alt - llvm, debug
 * x86_64-apple - llvm, debug
 * dist-aarch64-apple - llvm, debug

**Comment by alexcrichton** [comment]
For posterity, this had a pretty massive impact on CI times for the thre platforms that we disabled debug/llvm assertions for:


<img width="1162" alt="Screen Shot 2019-04-05 at 11 38 22 AM" src="https://user-images.githubusercontent.com/64996/55642905-5fc5fd80-5797-11e9-9a61-1c4eb7409f0f.png">

**Comment by aidanhs** [comment]
(currently waiting on a potential upgrade to our appveyor builder machines, maybe happening soonish)

**Comment by aidanhs** [comment]
Much has happened - we've migrated to azure in the meantime and our builds have got even slower, but there is potential to get 4 core machines there. Until then, this is still on hold.

**Comment by pietroalbini** [comment]
https://github.com/rust-lang/rust/pull/64517 also disabled LLVM assertions on the following builders:

* i686-gnu
* i686-gnu-nopt
* i686-msvc-1
* i686-msvc-2
* x86_64-msvc-cargo

**Comment by shepmaster** [comment]
Triage: Nothing has materially changed; we still don't have the time budget to re-enable these. Perhaps when the builders gain more cores.

**Comment by pietroalbini** [comment]
We'll probably want to re-enable those if we switch to GitHub Actions, as we have the time budget there.

**Comment by Mark-Simulacrum** [comment]
With GHA, we should now have capacity to do this on CI. I will be posting a PR doing so soon.

**Comment by Mark-Simulacrum** [comment]
Updated issue description, only macOS builders remain disabled.

**Comment by shepmaster** [comment]
Since this now deals with just the macOS builders, there's not much we can do at the moment. It would likely require some external factor, such as GitHub Actions drastically improving the performance of those machines.

**Comment by nikic** [comment]
The issue lists a few `dist-*` targets, but shouldn't assertions always be disabled for those, regardless of capacity?

**Comment by Mark-Simulacrum** [comment]
Yes, that seems correct - I think only alt dist builders are assert enabled today.
